myTrip/photo/views.py

```diff
@@ -1,40 +1,50 @@
 """Module generate view for photo requests."""
 
 import json
+
 from django.views.generic.base import View
 from django.http import JsonResponse, HttpResponse
 from .models import Photo
+from registration.models import CustomUser
 
 
 class PhotoView(View):
     """Class that handle HTTP requests."""
 
-    def get(self, request, photo_id):
-        """GET request handler.Get photo by photo_id"""
-        photo = Photo.get_by_id(photo_id)
-        if not photo:
-            return HttpResponse(status=404)
-        photo = photo.to_dict()
-        return JsonResponse(photo, status=200, safe=False)
+    def get(self, request, trip_id=None, checkpoint_id=None):
+        """GET request handler. Can return photo/photos ordered by given args."""
+        if trip_id and checkpoint_id:
+            photos = Photo.get_by_trip_id_and_checkpoint_id(trip_id, checkpoint_id)
+        photos = Photo.get_by_trip_id(trip_id)
```

Comment:
what if we have no photos given???
missing return in 1st if case + consider adding another if statement for better readability:
```
if trip_id and checkpoint_id:
    photos = Photo.get_by_trip_id_and_checkpoint_id(trip_id, checkpoint_id)
    data = [photo.to_dict() for photo in photos]
    return JsonResponse(data, status=200, safe=False)
if trip_id:
    photos = Photo.get_by_trip_id(trip_id)
    data = [photo.to_dict() for photo in photos]
    return JsonResponse(data, status=200, safe=False)
```